Do not override MacPorts deployment target.

Do not hardcode architectures when linking.
https://github.com/angband/angband/pull/507

Do not hide compile and link commands.

Use cp -R flag instead of deprecated -r flag.
https://github.com/angband/angband/pull/506
--- src/Makefile.osx.orig	2018-07-21 23:49:08.000000000 -0500
+++ src/Makefile.osx	2018-10-14 11:30:01.000000000 -0500
@@ -13,13 +13,14 @@
 # for consistency with older versions, use a lower case bundle ID
 BUNDLE_IDENTIFIER = org.rephial.angband
 
-ARCH = -arch i386 -arch x86_64
+ARCHS = i386 x86_64
+ARCHFLAGS = $(addprefix -arch ,$(ARCHS))
 WARNINGS = -W -Wall -Wno-unused-parameter -Wno-missing-field-initializers \
 	-Wunused-macros
 JUST_C = -std=c99
-OBJ_CFLAGS = -std=c99 -x objective-c -mmacosx-version-min=10.5
+OBJ_CFLAGS = -std=c99 -x objective-c
 CFLAGS = -g -I. $(WARNINGS) $(OPT) -DMACH_O_CARBON -DHAVE_MKSTEMP \
-	-fno-stack-protector $(ARCH)
+	-fno-stack-protector $(ARCHFLAGS)
 LIBS = -framework Cocoa
 # Fix for bug #1663: Set the deployment target via environment variable
 # for the final link command. See http://grauonline.de/wordpress/?p=71
@@ -61,23 +62,23 @@
 
 
 $(EXE).o: $(OBJS)
-	@printf "%10s %-20s\n" LD $@.x86_64
-	@$(LD) -r -arch x86_64 -o $@.x86_64 $(OBJS)
-	@printf "%10s %-20s\n" LD $@.i386
-	@$(LD) -r -arch i386 -o $@.i386 $(OBJS)
-	lipo -arch x86_64 $@.x86_64 -arch i386 $@.i386 -create -output $(PROGNAME).o
+	for A in $(ARCHS); do \
+		$(LD) -r -arch $$A -o $@.$$A $(OBJS); \
+		LIPO_ARGS="$$LIPO_ARGS -arch $$A $@.$$A"; \
+	done; \
+	lipo $$LIPO_ARGS -create -output $@
 
 izzywizzy: $(OBJS) izzywizzy.o
 
 $(EXE): $(EXE).o $(OSX_OBJS)
-	$(DEPLOYMENT_TARGET) $(CC) $(CFLAGS) $(LDFLAGS) -o $(EXE) $(EXE).o $(OSX_OBJS) $(LIBS) 
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $(EXE) $(EXE).o $(OSX_OBJS) $(LIBS)
 
 #
 # Clean up old junk
 #
 
 clean:
-	-rm -f $(OBJS) $(EXE) $(EXE).o.i386 $(EXE).o.x86_64 $(EXE).o $(OSX_OBJS)
+	-rm -f $(OBJS) $(EXE) $(addprefix $(EXE).o.,$(ARCHS)) $(EXE).o $(OSX_OBJS)
 
 
 #
@@ -85,11 +86,9 @@
 #
 
 main-cocoa.o : main-cocoa.m
-	@printf "%10s %-20s (objective-c)\n" CC $<
-	@$(CC) $(CFLAGS) $(OBJ_CFLAGS) -c -o $@ $<
+	$(CC) $(CFLAGS) $(OBJ_CFLAGS) -c -o $@ $<
 %.o : %.c
-	@printf "%10s %-20s\n" CC $<
-	@$(CC) $(JUST_C) $(CFLAGS) -c -o $@ $<
+	$(CC) $(JUST_C) $(CFLAGS) -c -o $@ $<
 
 
 #
@@ -151,7 +150,7 @@
 	@cp ../lib/customize/*.prf $(APPRES)/lib/customize
 
 	# 'optionally' install sound and graphics tiles, if present
-	-cp -r ../lib/tiles/ $(APPRES)/lib/tiles
+	-cp -R ../lib/tiles/ $(APPRES)/lib/tiles
 	-rm $(APPRES)/lib/tiles/Makefile $(APPRES)/lib/tiles/*/Makefile
 	-cp ../lib/sounds/*.mp3 $(APPRES)/lib/sounds
 	-cp ../lib/sounds/sound.cfg $(APPRES)/lib/sounds
